controllers/openstackcluster_controller.go

```diff
@@ -126,9 +127,13 @@ func reconcileDelete(ctx context.Context, log logr.Logger, client client.Client,
 		return reconcile.Result{}, err
 	}
 
-	if err = deleteBastion(log, osProviderClient, clientOpts, cluster, openStackCluster); err != nil {
+	deleted, err := deleteBastion(log, osProviderClient, clientOpts, cluster, openStackCluster)
+	if err != nil {
 		return reconcile.Result{}, err
 	}
+	if !deleted {
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
```

Comment:
Won't it make sense here to use a exponential backoff (`return reconcile.Result{}, nil`) to not "DDoS" the API?

Reply:
I've updated everything to use the controller's rate limiter. I've also customised the rate limiter to have a higher base delay. Without writing my own exponential backoff rate limiter (which I wasn't keen on doing) we're stuck with an exponent of 2, so I've set the base delay at 2 seconds. We will back off at intervals of 2s, 4s, 8s... up to a max of 1000s.